index.html

```diff
@@ -366,14 +369,27 @@ <h2><dfn>PerformanceObserverInit</dfn> dictionary</h2>
           <pre class="idl">
           dictionary PerformanceObserverInit {
             required sequence&lt;DOMString&gt; entryTypes;
-            boolean buffered = false;
           };
           </pre>
           <dl>
             <dt><dfn>entryTypes</dfn></dt>
             <dd>A list of entry types to be observed. The list MUST NOT be empty
             and types not recognized by the user agent MUST be ignored.</dd>
           </dl>
+          <p class="note">Whenever a new kind of <a>PerformanceEntry</a> is defined,
+            the <a>PerformanceObserverInit</a> dictionary should be augmented to
+            include a new optional <a>PerformanceEntryObserverOptions</a> dictionary
+            with the name of the new <a data-lt="PerformanceEntry.entryType">
+            entryType</a>.</p>
```

Comment:
We don't do this today for existing types. If we follow this recommendation, than we should do so.

That said, we've also tried to explicitly avoid enumerating "valid types" in Performance Timeline, as that makes a hard dependency for the perf timeline to be revved whenever we add a new type. Per discussion in https://github.com/w3c/performance-timeline/issues/77, the idea is to have other specs call into Perf Timeline to register available types. The logic here should then enumerate the dictionary members, compare against registered + known types and proceed accordingly. This also means that we ignore any unknown types in this algorithm.

WDYT, does that make sense?

Reply:
I'm not forcing PerformanceTimeline to be modified whenever a new entry type is added. I was thinking the new spec could do something along the lines of 'partial interface' to add the new dictionary corresponding to its entry type. We already "remove all unsupported types", which is a bit hand-wavy but I think it's clear what it means. If we want to make it a bit more formal, we can go with your registration approach, I just don't know if it adds much clarity or if it makes speccing new entry types easier.

Reply:
Ah, I'm with you now. I like that, that's much cleaner than I what I had in mind before. The one bit I'd change in this proposal is, instead of extending the `PerformanceObserverInit` we wanted + need to introduce `performance.entryTypes` and that's the interface the various specs should be extending. In turn, we can then filter passed options to `PerformanceObserverInit` against said interface.

WDYT, does that make sense? If you're on board with that.. the changes are coupled, and we would need to land the `performance.entryTypes` proposal first, and then rebase this against that.

Reply:
Sorry, I don't follow what you're suggesting here... `performance.entryTypes` will only help in knowing which are the entry types that are supported globally (implemented by the browser), right? So we still need a way to specify what filters can be used for each entry type, unless we want to support the same set of filters for all entry types? Even if we have the same set of filters on all entry types, we still need to augment `PerformanceObserverInit` with the dictionaries per entry type (I don't think we want to allow it to contain dictionaries like 'longtask' and 'mark' without specifying them anywhere explicitly). The only alternative I can think of that avoids this would be to have something like this:
```
dictionary PerformanceObserverInit {
  required sequence<DOMString> entryTypes;
  PerformanceEntryObserverOptions filters;
}
```
But this would force an observer to apply the same set of filters to all of the observed entry types that it is observing. This is kind of what we're trying to avoid here, since ideally we'd like entryType-specific parameters. If we don't have that, we might as well just keep adding the filters to `PerformanceObserverInit` directly. WDYT?